tables/array.py

```diff
@@ -348,7 +362,9 @@ def __next__(self):
             else:
                 return self.listarr    # Scalar case
 
-    def _interpret_indexing(self, keys):
+    def _interpret_indexing(
+            self, keys: Union[int, slice, list[Union[int, slice]], np.ndarray],
```

Comment:
Not sure if in this case we could use `ArrayLike`.
Probably your solution is sharper.

Reply:
Unfortunately, a `slice` is not `ArrayLike`. Because of this, I let the annotation as is.
If you want to have the change, I can change it, though.

Reply:
In this case the comment was about the possibility to use 
```python
keys: Union[int, slice, list[Union[int, slice]], ArrayLike],
```

in place of 

```python
keys: Union[int, slice, list[Union[int, slice]], np.ndarray],
```

Reply:
Ah, got it! Thanks for the input. Changed it to `ArrayLike`.